core/domain/exp_services.py

```diff
@@ -1909,14 +1909,26 @@ def regenerate_missing_stats_for_exploration(exp_id):
 
             try:
                 prev_interaction_id = (
-                    prev_exp.state_interaction_ids_dict[state_name])
+                    prev_exp.state_interaction_ids_dict[prev_state_name]
+                    if prev_state_name in prev_exp.state_interaction_ids_dict
+                    else None)
```

Comment:
Just to check, there are no issues with setting exp ID to None, right? (As in, is it used in a db call?)

Reply:
It's interaction id here. We aren't using it to check if the new and old states have the same interaction below. It shouldn't be an issue.